core/commands/storage.go

```diff
@@ -304,20 +312,31 @@ var storageUploadRequestChallengeCmd = &cmds.Command{
 		if err != nil {
 			return err
 		}
-		cid, err := cid2.Parse(chunkHash)
+		cid, err := cidlib.Parse(chunkHash)
 		if err != nil {
 			return err
 		}
-		sch, err := NewStorageChallenge(req.Context, n, api, cid)
-		if err != nil {
-			return err
+
+		var mutex = sync.Mutex{}
```

Comment:
This mutex is not shared! Must use a shared mutex from this session.

Can you perform the following logic all in a `ss.SetChallenge` so you can re-use ss's mutex?